plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -632,6 +634,9 @@ public Collection<Object> createComponents(
             .getClusterSettings()
             .addSettingsUpdateConsumer(MLCommonsSettings.ML_COMMONS_RAG_PIPELINE_FEATURE_ENABLED, it -> ragSearchPipelineEnabled = it);
 
+        SdkClient sdkClient = new SdkClient();
+        sdkClient.setImplementation(new XContentClient(client, xContentRegistry));
+
```

Comment:
Add a todo to initialize client based on settings. There should be support to set different implementations.

Reply:
There should be, but probably not in the plugin (see previous comment on why a concrete class and not an interface).  I think the implementation will be determined elsewhere and the plugin will just be handled an initialized object.

For development/testing it might be nice to have a setting-based switch between two different implementations on-the-fly.

Reply:
It might be better to handle plugin dependencies with in the plugin rather than relying on another component. We might be able to make use of `createGuiceModules` to initialize dependencies within plugin.